### PR TITLE
psoc-programmer: init at 2014-11-26

### PR DESCRIPTION
**Pull request by lowfatcomputing**
- Added package for Free Software Cypress PSoC toolchain.

I'd be interested in feedback or alternative suggestions (if any) about the psoc-prog wrapper script I created because of the long config directory path.

I used fetchgit instead of fetchFromGitHub because fetchFromGitHub can't pull commits from branches other than master as far as I know.

### CODE REVIEW

File: pkgs/tools/misc/psoc-programmer/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    install -Dm644 config/nm.hex  $out/etc/psoc-programmer/nm.hex
+
+    # docs
+    install -Dm644 README.md $out/share/doc/psoc-programmer/README.md
+    install -Dm644 TODO $out/share/doc/psoc-programmer/TODO
+    install -Dm644 docs/* $out/share/doc/psoc-programmer/
+  '';
+
+  meta = {
+    description = "Open source tools to manipulate hex files and program a PSoC5 via FX2 USB interface.";
```
Comment (jagajaga) [inline pkgs/tools/misc/psoc-programmer/default.nix]: Please remove dot at the end the description.

**Comment by lowfatcomputing** [comment]
> I'd be interested in feedback or alternative suggestions (if any) about the psoc-prog wrapper script I created because of the long config directory path.

I'm concerned that the name `prog` will have conflicts with other packages. I'm going to go modify `prog` later tonight to read config from a default directory unless otherwise specified on cli and rename the executable to `psoc-prog`. Upstream is inactive.

**Comment by pSub** [comment]
@lowfatcomputing any updates on this?

**Comment by vcunat** [comment]
Ping. BTW, `fetchFromGitHub` can fetch *any* revision (and the general version `fetchzip` fetches various archives).

**Comment by jagajaga** [comment]
Please ping us when you finish :)

**Comment by Profpatsch** [comment]
(triage) @lowfatcomputing What’s the status?

**Comment by grahamc** [comment]
I'd suggest moving to close this PR until renewed interest comes along.

**Comment by joachifm** [comment]
Closing as abandoned.

**Comment by lowfatcomputing** [comment]
Yeah... apologies for not responding sooner. I have a NixOS installation again, but I have abandoned work on `psoc-programmer` because `openocd` has support for PSoC4, and PSoC5 support is in the works. `openocd` will anyway be the way to go in the future, so there is less reason to continue developing psoc-programmer. Still good as example code though (for FX2LP programmer support, although most likely people are now using the kitprog programmer with the CY8CKIT-059 devkit since that became available).

**Pull request closed without merging**